### PR TITLE
fix: Three clear options for persons on events

### PR DESCRIPTION
**Pull request by timgl**
## Problem

We have too many options for persons on events, and they're confusing. What really matters is a. whether person_id needs to be overridden or not, and b: whether person properties need to be joined

## Changes
Current options:
- Team level setting aka V1
- Feature flag aka V2 https://us.posthog.com/project/2/feature_flags/8523

This PR adds:
1. Person ID: no override, properties: on events
  - Put everyone that currently manually set the setting on to this
2. Person ID: override, properties: join
  - Migrate everyone else to this
3. Person ID: override, properties: on events
  - Put everyone currently on v2 on this (their data will change)
  - Goal is to eventually get everyone on this

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

**Review by tkaemming — APPROVED** [review]

File: posthog/models/team/team.py [hunk 1]
```diff
@@ -362,23 +366,26 @@ def _person_on_events_v2_querying_enabled(self) -> bool:
...
+                    "organization": {
+                        "id": str(self.organization_id),
+                        "created_at": self.organization.created_at,
+                    }
+                },
+                only_evaluate_locally=True,
+                send_feature_flag_events=False,
+            )
+
+        # on self-hosted, use the instance setting
```
[the remark this answers is not in the document]
Comment (timgl) [reply posthog/models/team/team.py:381]: This is still true no?

File: posthog/hogql/modifiers.py [hunk 2]
```diff
@@ -21,10 +16,7 @@ def create_default_modifiers_for_team(
         modifiers = modifiers.model_copy()
 
     if modifiers.personsOnEventsMode is None:
-        if team.person_on_events_v3_querying_enabled:
-            modifiers.personsOnEventsMode = PersonsOnEventsMode.v3_enabled
-        else:
-            modifiers.personsOnEventsMode = team.person_on_events_mode
+        modifiers.personsOnEventsMode = team.person_on_events_mode or PersonOnEventsMode.DISABLED
```
[the remark this answers is not in the document]
Comment (timgl) [reply posthog/hogql/modifiers.py:19]: I've added the missing option here, assuming you'll pick up actually making that option do what it says?

File: posthog/models/team/team.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tkaemming) [reply posthog/models/team/team.py:381]: I don't think so for this one — there's no fallthrough to a `get_instance_setting` (contrast with line 343 above) unless that happens somewhere else that I'm forgetting about, or maybe interpreting the comment differently?

File: posthog/hogql/modifiers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tkaemming) [reply posthog/hogql/modifiers.py:19]: > assuming you'll pick up actually making that option do what it says

Yep, that was my plan. (I was assuming that would happen as a separate PR and that you weren't waiting for me to add those changes into this one -- just to make sure we're on the same page here.)

File: posthog/models/team/team.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (timgl) [reply posthog/models/team/team.py:380]: ah got confused, yes you're right!

**Pull request merged**